### PR TITLE
fixes #13

### LINKED ISSUE #13: Add home reset delay preference
Currently, the home reset delay is hardcoded to about 3 seconds after pressing any switch. However, an inverse scanning user would find it really hard to access Tecla as the app would continually send them Home when trying to do anything. Therefore, the home reset event has been disabled for inverse scanning until a preference is exposed.

### PR DESCRIPTION
**Pull request by vishesh04**
Added preference to set home reset time delay.

### CODE REVIEW

**Comment by ijdoc** [comment]
Great work! seems you recycled the dialog from other preferences. I have some comments about the timeout values chosen.
1. You provide options in 3 second increments but I think it makes sense to provide higher increments on higher values.
2. We have some users with custom builds with up to 32 seconds in timeout. I think that should be the upper level.
3. I don't think we want to have a "Never" option, specially since that is the only way to continue using the phone in cases when the Android OS is uncooperative
If you make these changes we can merge the pull request!

**Comment by vishesh04** [comment]
Changed slider values.Now higher values have higher increment and never option has been removed

File: source/res/values/strings.xml [hunk 1]
```diff
@@ -250,6 +250,22 @@
 		<item>Never</item>
 	</string-array>
 
+    <!-- Option to set full reset timeout -->
+	<string name="full_reset_timeout">Full Reset Timeout</string>
+	<string name="full_reset_timeout_summary">Reset to Home after this time delay</string>
+    <string name="full_reset_timeout_dlg_title">Set Full Reset Timeout</string>
```
Comment (ijdoc) [inline source/res/values/strings.xml]: Should be the same as 254 above

File: source/src/ca/idi/tekla/util/Persistence.java [hunk 2]
```diff
@@ -20,17 +20,20 @@
...
 	public static final String PREF_SELF_SCANNING = "self_scanning";
 	public static final String PREF_INVERSE_SCANNING = "inverse_scanning";
 	public static final String PREF_SCAN_DELAY_INT = "scan_delay_int";
+	public static final long DEFAULT_FULL_RESET_TIMEOUT = 3000; //millisecond
 	public static final int DEFAULT_SCAN_DELAY = 1000;
 	public static final int MAX_SCAN_DELAY = 3000;
 	public static final int MIN_SCAN_DELAY = 250;
 	public static final int AUTOHIDE_NULL = -999;
 	public static final int NEVER_AUTOHIDE = -1;
+	public static final int NEVER_FULL_RESET = -1;
```
Comment (ijdoc) [inline source/src/ca/idi/tekla/util/Persistence.java]: Is this used anymore? Should we remove it?

Comment (vishesh04) [inline source/src/ca/idi/tekla/util/Persistence.java]: Hi I have made the following changes and pushed them to my fork

_in persitance.java_
We no longer need NEVER_FULL_RESET,I have removed it.
We need PREF_FULL_RESET_TIMEOUT to create getter and setter for this
preference
DEFAULT_FULL_RESET_TIMEOUT has been set to 3 seconds.

_in strings.xml_
*\* do you mean all strings ( line 254,255,256) must be same as "Full Reset
Timeout".I have made these changes.

Please comment on these changes

On Mon, Apr 16, 2012 at 8:54 AM, Jorge Silva <
reply@reply.github.com

> wrote:
> 
> > ```
> >   public static final int DEFAULT_SCAN_DELAY = 1000;
> >   public static final int MAX_SCAN_DELAY = 3000;
> >   public static final int MIN_SCAN_DELAY = 250;
> >   public static final int AUTOHIDE_NULL = -999;
> >   public static final int NEVER_AUTOHIDE = -1;
> > ```
> > -     public static final int NEVER_FULL_RESET = -1;
> 
> Is this used anymore? Should we remove it?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jorgesilva/TeclaAccess/pull/67/files#r686747

## 

Vishesh Singhal
Electrical & Electronics
BITS-PILANI

**Pull request merged**